File: kouncil-backend/src/main/java/com/consdata/kouncil/TopicMessages.java
package com.consdata.kouncil;

import lombok.Builder;
import lombok.Data;
import lombok.Singular;

import java.util.List;
import java.util.Map;

@Data
@Builder
public class TopicMessages {

	@Singular
	private List<Message> messages;

	private Map<Integer, Long> partitionOffsets;

	private Map<Integer, Long> partitionEndOffsets;

	private Long totalResults;
}

File: kouncil-frontend/apps/kouncil/src/app/topic/topic-messages.ts
import {MessageData} from '@app/message-data';

export class TopicMessages {
  constructor(public messages: MessageData[],
              public partitionOffsets: { [key: number]: number },
              public partitionEndOffsets: { [key: number]: number },
              public totalResults: number) {

  }
}
